cmd/root.go

```diff
@@ -29,13 +30,46 @@ func Execute() {
 	rootCmd.Execute()
 }
 
+func getCacheDirFromTFConfig() (string, error) {
+	tfCfgPath, ok := os.LookupEnv("TF_CLI_CONFIG_FILE")
+	if !ok {
+		// The file not existing is not an error
+		return "", nil
+	}
+
+	data, err := ioutil.ReadFile(tfCfgPath)
+	if err != nil {
+		return "", err
+	}
+
+	var cfg struct{
+		PluginCacheDir string `hcl:"plugin_cache_dir"`
+	}
+	err = hcl.Unmarshal(data, &cfg)
+	if err != nil {
+		return "", err
+	}
+
+	return cfg.PluginCacheDir, nil
+}
+
 func init() {
 	cobra.OnInitialize(initConfig)
 
+	cacheDir := "$HOME/.terraform.d/plugin-cache"
```

Comment:
I think we should use [path/filepath](https://pkg.go.dev/path/filepath) like the previous default value to ensure cross-platform:

```go
cacheDir := filepath.Join(os.ExpandEnv("$HOME"), "/.terraform.d/plugin-cache")
```

Reply:
Since we're hard-coding the /:es for the "inner" parts already, I thought that didn't actually help. But now I read the source of `Join`, and it apparently post-processes the result with `Clean`, which replaces slashes if required.

So perhaps we can do `filepath.Clean` before using the result instead?

Reply:
Great, I think that's a good idea!